pages/report-hazard/script.js

```diff
@@ -1,102 +1,118 @@
+import Map from '../../assets/models/Map.js'
 
 //Variable Declaration
-const currentReport = new ReportForm();
-let position = Map.DEFAULT_LOCATION;
-let map = null;
+const currentReport = new ReportForm()
+let position = Map.DEFAULT_LOCATION
+let map = null
 
 /**
  * Page Init
  */
 
 window.onload = function () {
-  displayCurrentSection();
-  window.addEventListener("hashchange", displayCurrentSection);
+	displayCurrentSection()
+	window.addEventListener('hashchange', displayCurrentSection)
 
-  // Loads the map even if the user has not accepted the permissions
-  map = new Map(position);
-  map.setMarkerOnMap(position.latitude, position.longitude, "You", {
-    draggable: true,
-  }); //TODO: Consult with design the message of the marker
+	// Loads the map even if the user has not accepted the permissions
+	map = new Map(position)
+	map.setMarkerOnMap(position.latitude, position.longitude, 'You', {
+		draggable: true,
+	}) //TODO: Consult with design the message of the marker
 
-  //Override the current location if the user accepts the permissions
-  loadGeolocation();
-};
+	//Override the current location if the user accepts the permissions
+	loadGeolocation()
+}
 
 const displayCurrentSection = () => {
-  const allPages = document.querySelectorAll("section.page");
-
-  const pageId = location.hash ? location.hash : "#select-location";
-  for (let page of allPages) {
-    if (pageId === "#" + page.id) {
-      page.style.display = "block";
-    } else {
-      page.style.display = "none";
-    }
-  }
-};
+	const allPages = document.querySelectorAll('section.page')
+
+	const pageId = location.hash ? location.hash : '#select-location'
+	for (let page of allPages) {
+		if (pageId === '#' + page.id) {
+			page.style.display = 'block'
+		} else {
+			page.style.display = 'none'
+		}
+	}
+}
 
 const loadGeolocation = async () => {
-  position = await Map.getCurrentLocation();
-  map.setMarkerOnMap(position.latitude, position.longitude, "You", {
-    draggable: true,
-  });
-};
+	position = await Map.getCurrentLocation()
+	map.setMarkerOnMap(position.latitude, position.longitude, 'You', {
+		draggable: true,
+	})
+}
 
 /**
  * Step 1: Location
  */
 
 if (map) {
-  map.on("click", onSelectLocation);
+	map.on('click', onSelectLocation)
 }
 
 const onSelectLocation = (event) => {
-  map.removeLayer(marker);
-  map.setMarkerOnMap(event.latlng.lat, event.latlng.lng, "Location selected", {
-    draggable: true,
-  });
-
-  currentReport.location = {
-    lat: event.latlng.lat,
-    lng: event.latlng.lng,
-    address: "Fake address", //TODO: Get address
-  };
-};
+	map.removeLayer(marker)
+	map.setMarkerOnMap(event.latlng.lat, event.latlng.lng, 'Location selected', {
+		draggable: true,
+	})
+
+	currentReport.location = {
+		lat: event.latlng.lat,
+		lng: event.latlng.lng,
+		address: 'Fake address', //TODO: Get address
+	}
+}
 
 /**
  * Step 2: Category List
  */
+
+async function getCategory() {
```

Comment:
Current
```
async function getCategory() {
	const baseURL =
		'https://enchanting-llama-6664aa.netlify.app/.netlify/functions/hazard-category'

	let res = await fetch(baseURL)
	let data = await res.json()

	for (let i = 0; i < data.data.length; i++) {
		catego = data.data[i].name
		var catego = document.getElementById('category' + (i + 1))
		catego.innerHTML = data.data[i].name
	}
}
getCategory()
```

- If we are retrieving a list the function should be getCategories
- Always wrap your code inside a try-catch
- If you are doing this data.data[i], more than once define a variable, e.g. const category = data.data[i]
- instead of doing this `let data = await res.json()` and access data like this: `data.data`, you can do it directly with `const { data } = await res.json()`, notice how this variable doesn't change, so it should be defined as a constant (with const instead of let)
- Use API_URL from constants files.
- As a general note, notice how we work with the brake lines, so the code is more readable and you have several chunks of code.
- `catego` is not a good variable name, use full words in these cases
- An important notice of the use of the id, the id is the value that we are going to need to send at the end when we save the information, if we don't have a way to retrieve the id of the category selected, we cannot complete the request at the end.

```suggestion
const getCategories = async () => {
	try {
		let response = await fetch(`${API_URL}/hazard-category`)
		let { data } = await response.json()

		const content = document.getElementById('hazard-category-content')

		for (let i = 0; i < data.length; i++) {
			const category = data[i]
			const div = document.createElement('div')

			const radio = document.createElement('input')
			radio.setAttribute('type', 'radio')
			radio.setAttribute('name', 'categoryRadioBtn')
			radio.setAttribute('id', `category-${category.id}-radio`)
			radio.setAttribute('value', category.id)
			radio.addEventListener('change', (event) => {
				window.location.href = '#hazard-type'
				currentReport.categoryId = event.target.value
			})

			const label = document.createElement('label')
			label.setAttribute('id', `category-${category.id}-label`)
			label.setAttribute('for', `category-${category.id}-radio`)
			label.innerHTML = category.name

			div.appendChild(radio)
			div.appendChild(label)

			content.appendChild(div)
		}
	} catch (error) {
		//TODO: display error message
	}
}
getCategories()
```

Reply:
done